README.md

```diff
@@ -424,7 +426,51 @@ Assumes database resolution is done by adding prefixes. If database resolution i
      where h.date = '2016-05-13'
        and h.hour = 1
     ;
-    
+
+## Database Name Mapping
+NOTE: mapping names adds an extra layer of abstraction and we advise to use this as a temporary migration solution only. It becomes harder to debug where a virtual (remapped) table actually is coming from.
+
+Waggle Dance allows one to refer to databases by names that are different to what they are defined in the Hive metastore via a `database-name-mapping` configuration. This feature can be useful when migrating data from existing databases into different environments.
+To clarify how this feature could be used, below is an example use case:
+As part of a data migration we have decided that we want to store all Hive tables related to hotel bookings in a database called `booking`. However we have legacy data lakes that have tables in this domain but they are stored in databases with names that don't match this new standard, e.g. 'datawarehouse'. To ease migration we want to expose the 'datawarehouse' db as both the original name and the booking name. This way consumers can start using the new name while producers migrate their scripts from the old name or visa versa. When the migration is done the mapping can be removed and the database renamed.
+
+So in this example we have a Data lake X which federates tables from another Data lake Y.
+The desired end result is to show a 'datawarehouse' database and a 'booking' database in X which is proxied to the same 'datawarehouse' database in Y.
+
+X: _show databases;_
+
+    datawarehouse (proxies to y - datawarehouse)
+    booking (proxies to y - datawarehouse)
+
+Y: _show databases;_
+
+    datawarehouse
+
+To achieve a unified view of all the booking tables in the different databases (without actually renaming them in Hive) we can configure Waggle Dance in X to map from the old to the new names like so:
+
+    federated-meta-stores:
+      - remote-meta-store-uris: thrift://10.0.0.1:9083
+        name: y
+        mapped-databases:
+        - datawarehouse
+        database-name-mapping:
+          datawarehouse: booking
+
+Note: Both the 'datawarehouse' and the mapped name 'booking' is show in X, so the mapping adds an additional virtual database mapping both to the same remote database. You can only map one extra name and you cannot map different databases to the same name.
```

Comment:
```suggestion
Note: Both the 'datawarehouse' name and the mapped name 'booking' are shown in X, so the mapping adds an additional virtual database mapping to the same remote database. You can only map one extra name and you cannot map different databases to the same name.
```

Reply:
Just out of curiosity, is the restriction `You can only map one extra name` a technical restriction, or just something that was picked as a limit?  It seems from your other changed code using `transformOutboundDatabaseNameMultiple` that it returns a List, which would imply any number of mappings for a database.  But I freely admit I don't understand the internals of Waggledance.

Reply:
it's me being a little lazy and a technical thingy, so in yaml it's a map so I'd have to make this into a list of mappings. It can be done wasn't sure if that's needed. Happy to change it around though :)

Reply:
We don't have a use-case for that, so it sounds like too much trouble for something with little utility.  Was more just curious.  Thanks!